R/03_lda_analysis.qmd

```diff
@@ -0,0 +1,292 @@
+lda_fits_low_res %>%
+  DHSCconsultations::write_lda_fit(
+    here::here(
+      config$general_params$output_dir,
+      "lda_fit"
+    ), 
+    prefix = "low_res"
+  )
+```
+
+### Number of topics per question - high resolution scan
+
+Perform the high resolution scan of topic numbers.
+
+#### Set high resolution scan ranges
+
+These ranges have been picked as they cover the points on either side of the initial maximum log-likelihood found using the low resolution scan.
+
+```{r}
+topic_number_high_res <- list(
+  "Sense & Sensibility" = seq(10, 30, 2),
+  "Pride & Prejudice" = seq(10, 30, 2),
+  "Mansfield Park" = seq(10, 30, 2),
+  "Emma" = seq(10, 30, 2),
+  "Northanger Abbey" = seq(10, 30, 2),
+  "Persuasion" = seq(10, 30, 2)
+)
+```
+
+#### Run LDA scan
+
+Run the model across the low resolution range of topic numbers provided above.
+
+Note - the below can take significant time to complete for a large number of responses.
+
+```{r}
+lda_fits_high_res <- 
+  DHSCconsultations::fit_lda(
+    responses,
+    "response_id",
+    topic_number_high_res,
+    lda_seed,
+    clean_text,
+    glossary_words,
+    stop_words,
+    stem_word_exceptions
+  )
+```
+
+#### Check model convergence
+
+Check convergence at higher resolution.
+
+```{r}
+lda_fits_high_res %>%
+  DHSCconsultations::view_lda_fit_convergence()
+```
+
+#### Identify final topic number
+
+Generate graphs for the log-likelihood across the different number of topics for each question. The final topic number is the smallest number that achieves maximum log-likelihood.
+
+Note - it is a good idea to be conservative and select more topics if there is any ambiguity as topics will be manually checked.
+
+```{r}
+lda_fits_high_res %>%
+  DHSCconsultations::view_lda_fit_loglikelihood()
+```
+
+#### Save output
+
+```{r}
+lda_fits_high_res %>%
+  DHSCconsultations::write_lda_fit(
+    here::here(
+      config$general_params$output_dir,
+      "lda_fit"
+    ), 
+    prefix = "high_res"
+  )
+```
+
+## Final model
+
+Based on the above fits select the final number of topics for each questions and run the LDA.
+
+### Number of topics
+
+```{r}
+topic_number <- list(
+  "Sense & Sensibility" = 18,
+  "Pride & Prejudice" = 22,
+  "Mansfield Park" = 22,
+  "Emma" = 24,
+  "Northanger Abbey" = 14,
+  "Persuasion" = 16
+)
+```
+
+### Run LDA
+
+Run the model across topic numbers provided above.
+
+Note - the below can take significant time to complete.
+
+```{r}
+lda_final <- 
+  DHSCconsultations::run_lda(
+    responses,
+    "response_id",
+    topic_number,
+    lda_seed,
+    clean_text,
+    glossary_words,
+    stop_words,
+    stem_word_exceptions
+  )
+```
+
+### Check model convergence
+
+Check convergence of final fit. If there is a significant difference between the final value of the log-likelihood and the mean then you will need to re-run the model using a greater burn-in period and/or number of iterations.
+
+```{r}
+lda_final %>%
+  DHSCconsultations::view_lda_convergence()
```

Comment:
this now called view_lda_fitter() ? If so, it outputs the graph plus errors:
Error in unit(rep(0, TABLE_COLS * dims[2]), "null") : 
  'x' and 'units' must have length > 0